### DISCUSSION TITLE
Pagination Ideas

### DISCUSSION THREAD

**Discussion by jnebab** [web3phl/directory#29] [Brainstorm]
Hi guys, we need your input on what pagination you think is best for the directory. We are leaning toward the `Load More` button but ` infinite scrolling` is also a good option, especially for mobile users. The only caveat for` infinite scrolling` and `load more` is that it might get too long for desktop users. Here are more options just in case:

Numbered Pagination
<img width="835" alt="image" src="https://user-images.githubusercontent.com/6265768/185668096-b3eb7cac-c4ee-44d9-8d4f-0e08b759acd1.png">

Previous and Next Pagination
<img width="835" alt="image" src="https://user-images.githubusercontent.com/6265768/185668254-1e9dae16-006d-4a31-8e03-622d5ba0b83c.png">

Previous and Next Pagination with current  page and total items
<img width="835" alt="image" src="https://user-images.githubusercontent.com/6265768/185668369-b3b04b9f-8b96-4e8a-a74c-00864adcc90a.png">

Load More button only
<img width="835" alt="image" src="https://user-images.githubusercontent.com/6265768/185668423-f6aa1d8b-7ec9-4966-ab63-7fe317669892.png">

Infinite Loading on Scroll down
<img width="835" alt="image" src="https://user-images.githubusercontent.com/6265768/185669128-25fe77db-e1e9-4565-8ede-bdaaee044b82.png">

https://react-infinite-scroll-component.netlify.app/?path=/story/components--infinitescroll

---

**UPDATE**: submitted as feature request here: #41

**Comment by warengonzaga** [comment]
@jnebab I prefer the first one. 👀 Not sure with the other community if they agree on it.

**Comment by kquirapas** [comment]
@jnebab Would there be any issue if we do the following:

1. 3 for desktop users
2. Infinite scroll for mobile?

**Reply by jnebab** [reply]
I believe it can work, it will just take a longer time to implement.

**Comment by RAcbd** [comment]
1 and 3 are pretty standard for desktop users. Infinite scrolling is okay on mobile as long as we also implement a filter or search.

**Reply by jnebab** [reply]
then we can do 3 for desktop and infinite for mobile. we can create an issue for his feature @warengonzaga. someone might want to implement it. as for filter and search. what do you think we prioritize first? filter/search or pagination?

**Reply by warengonzaga** [reply]
I believe @jnebab I'll create an issue for pagination and search filter therefore we can attach it to any PR we'll receive. Then tagging this idea back here for reference.
